src/analyses/variable-sensitivity/abstract_object.cpp

```diff
@@ -189,7 +188,16 @@ abstract_object_pointert abstract_objectt::expression_transform(
   }
 
   exprt simplified=simplify_expr(constant_replaced_expr, ns);
-  return environment.abstract_object_factory(simplified.type(), simplified, ns);
+  if(simplified.is_constant())
+  {
+    return environment.abstract_object_factory(
+            simplified.type(), simplified, ns);
+  }
+  else
+  {
+    // if our final expression isn't a constant, just return top
+    return environment.abstract_object_factory(expr.type(), ns, true, false);
+  }
 }
```

Comment:
Fix looks good and is (correctly) in the default abstract transformer, rather than the simplify code as I had presumed it would be.  If I understand the fix correctly then the following should *not* verify before but will after this patch set has been applied:

```
int x = nondet_int();
assert(x * 0 == 0);
```